code/edge/src/test/java/com/adobe/marketing/mobile/EdgeHitProcessorTests.java

```diff
@@ -1287,6 +1217,256 @@ public void setLocationHint(final String hint, final int ttlSeconds) {
+			getExperienceEventWithConfig("testDatastreamId", null),
+			edgeConfig,
+			identityMap
+		)
+			.toDataEntity();
+		// test
+		hitProcessor =
+			new EdgeHitProcessor(
+				mockNetworkResponseHandler,
+				mockEdgeNetworkService,
+				mockNamedCollection,
+				mockSharedStateCallback,
+				new EdgeStateCallback() {
+					@Override
+					public Map<String, Object> getImplementationDetails() {
+						return implementationDetails;
+					}
+
+					@Override
+					public String getLocationHint() {
+						return null;
+					}
+
+					@Override
+					public void setLocationHint(final String hint, final int ttlSeconds) {
+						// not called by hit processor
+					}
+				}
+			);
+		assertProcessHitResult(dataEntity, true);
+
+		ArgumentCaptor<String> urlCaptor = ArgumentCaptor.forClass(String.class);
+		ArgumentCaptor<String> payloadCaptor = ArgumentCaptor.forClass(String.class);
+		verify(mockEdgeNetworkService, times(1))
+			.doRequest(
+				urlCaptor.capture(),
+				payloadCaptor.capture(),
+				ArgumentMatchers.anyMap(),
+				any(EdgeNetworkService.ResponseCallback.class)
+			);
+
+		// Assert that the url contains the overridden datastreamId
+		assertTrue(urlCaptor.getValue().contains("configId=testDatastreamId"));
+
+		JSONObject expectedSdkConfigJSON = new JSONObject(
+			"{\"sdkConfig\":{" + "\"datastream\":{" + "\"original\":\"works\"" + "}" + "}" + "}"
+		);
+
+		assertTrue(payloadCaptor.getValue().contains("sdkConfig"));
+		JSONObject requestJson = new JSONObject(payloadCaptor.getValue());
+		assertNotNull(requestJson);
+
+		JSONObject metaJson = requestJson.getJSONObject("meta");
+		JSONAsserts.assertExactMatch(expectedSdkConfigJSON, metaJson);
+	}
+
+	@Test
+	public void testProcessHit_experienceEvent_whenDatastreamConfigOverridesSet_sendsConfigOverrides()
+		throws Exception {
+		mockNetworkServiceResponse("https://test.com", new RetryResult(EdgeNetworkService.Retry.NO));
+
+		Map<String, Object> configOverrides = new HashMap<>();
+		configOverrides.put("key", "value");
+
+		DataEntity dataEntity = new EdgeDataEntity(
+			getExperienceEventWithConfig(null, configOverrides),
+			edgeConfig,
+			identityMap
+		)
+			.toDataEntity();
+		// test
+		hitProcessor =
+			new EdgeHitProcessor(
+				mockNetworkResponseHandler,
+				mockEdgeNetworkService,
+				mockNamedCollection,
+				mockSharedStateCallback,
+				new EdgeStateCallback() {
+					@Override
+					public Map<String, Object> getImplementationDetails() {
+						return implementationDetails;
+					}
+
+					@Override
+					public String getLocationHint() {
+						return null;
+					}
+
+					@Override
+					public void setLocationHint(final String hint, final int ttlSeconds) {
+						// not called by hit processor
+					}
+				}
+			);
+		assertProcessHitResult(dataEntity, true);
+
+		ArgumentCaptor<String> payloadCaptor = ArgumentCaptor.forClass(String.class);
+		verify(mockEdgeNetworkService, times(1))
+			.doRequest(
+				anyString(),
```

Comment:
Can you also validate that the config ID in the URL is as expected and not overridden?